erpnext/subcontracting/doctype/subcontracting_order/subcontracting_order.js

```diff
@@ -138,6 +138,26 @@ frappe.ui.form.on('Subcontracting Order', {
 	}
 });
 
+frappe.ui.form.on('Subcontracting Order Supplied Item', {
+	sourced_by_supplier: function (frm, cdt, cdn) {
+		let row = locals[cdt][cdn];
+		row.rate = row.amount = 0;
+		if (frm.doc.items && frm.doc.supplied_items) {
```

Comment:
Checks in JS work slightly differently than Python. An empty array is considered `true`-ish.

You wanna check for the length specifically:

```suggestion
		if (frm.doc.items.length > 0 && frm.doc.supplied_items.length > 0) {
```